index-advisor.md

```diff
@@ -0,0 +1,207 @@
+
+```sql
+Recommend Index Set <option> = <value>;
+Recommend Index Show;
+```
+
+There are four configurable options available, detailed below:
+1. timeout: Specifies the time limit for running the `RECOMMEND INDEX` command.
+2. max_num_index: Defines the maximum total number of indexes to include in the result of `RECOMMEND INDEX`.
+3. max_index_columns: Sets the maximum number of columns allowed in multi-column indexes in the result.
+4. max_num_query: Specifies the maximum number of queries to select from the statement summary workload.
+
+Users can view the current settings using the `RECOMMEND INDEX SHOW` command. Below is an example that displays the current option values and demonstrates how to modify the timeout option:
+
+```sql
+mysql> recommend index show;
++-------------------+-------+---------------------------------------------------------+
+| option            | value | description                                             |
++-------------------+-------+---------------------------------------------------------+
+| max_num_index     | 5     | The maximum number of indexes to recommend.             |
+| max_index_columns | 3     | The maximum number of columns in an index.              |
+| max_num_query     | 1000  | The maximum number of queries to recommend indexes.     |
+| timeout           | 30s   | The timeout of index advisor.                           |
++-------------------+-------+---------------------------------------------------------+
+4 rows in set (0.00 sec)
+
+mysql> recommend index set timeout='20s';
+Query OK, 1 row affected (0.00 sec)
+```
+
+This example shows how users can inspect and update the index advisor's settings to fine-tune its behavior for their workloads.
+
+## Limitations
+
+Here are some current limitations of the index recommendation feature, which we plan to address in the future:
+1. It does not support prepared statements, meaning `RECOMMEND INDEX RUN` cannot recommend indexes for queries executed through the `Prepare` and `Execute` protocol.
+2. It does not provide recommendations for deleting indexes. We need to merge the removing index logic (see below) to the `RECOOEND` comamnd in the future. 
+3. A UI for the Index Advisor will be available in the future.
+
+# Removing Unused Indexes
+TiDB provides two system views/tables to help users identify inactive indexes in their workload. Users can either mark such indexes as invisible as a transitional state before dropping them or drop them right away. 
+
+## View sys.schema_unused_indexes
+
+The `sys.schema_unused_indexes` view identifies indexes that have not been used since the startup of all TiDB instances. The view is defined based on system tables that have schema, table and column information. The view provides the full specification for the index including index, table and schema names. Users can query this view and decide on making indexes invisible or deleting them. 
+
```

Comment:
```suggestion

> **Warning:**
>
> As this view shows the unused indexes since last startup of all TiDB instances, please make sure the TiDB instances are alive long enough. Otherwise, it could show false candidates in case certain workloads are not included. 

```